src/main/kotlin/fi/metatavu/ikioma/rest/V1ApiImpl.kt

```diff
@@ -1,65 +1,135 @@
 package fi.metatavu.ikioma.rest
 
-import fi.metatavu.ikioma.email.EmailController
+import fi.metatavu.ikioma.controllers.EmailController
+import fi.metatavu.ikioma.controllers.KeycloakController
+import fi.metatavu.ikioma.controllers.PaymentController
+import fi.metatavu.ikioma.controllers.PrescriptionRenewalController
 import fi.metatavu.ikioma.email.api.api.spec.V1Api
-import fi.metatavu.ikioma.email.api.spec.model.Email
-import org.apache.commons.validator.routines.EmailValidator
+import fi.metatavu.ikioma.email.api.spec.model.PaymentStatus
+import fi.metatavu.ikioma.email.api.spec.model.PrescriptionRenewal
+import fi.metatavu.ikioma.rest.translate.PrescriptionRenewalTranslator
 import org.slf4j.Logger
-import java.util.concurrent.TimeUnit
+import java.lang.IllegalArgumentException
+import java.util.*
 import javax.annotation.security.RolesAllowed
 import javax.enterprise.context.RequestScoped
 import javax.inject.Inject
+import javax.transaction.Transactional
 import javax.ws.rs.core.Response
 
 
 /**
  * V1 API
  */
 @RequestScoped
+@Transactional
 class V1ApiImpl : V1Api, AbstractApi() {
 
     @Inject
     private lateinit var emailController: EmailController
 
+    @Inject
+    private lateinit var prescriptionController: PrescriptionRenewalController
+
+    @Inject
+    private lateinit var paymentController: PaymentController
+
+    @Inject
+    private lateinit var prescriptionRenewalTranslator: PrescriptionRenewalTranslator
+
+    @Inject
+    private lateinit var keycloakController: KeycloakController
+
     @Inject
     private lateinit var logger: Logger
 
-    @RolesAllowed(value = [ UserRole.PATIENT.name ])
-    override fun createEmail(email: Email): Response {
-        loggedUserId ?: return createUnauthorized("Unauthorized")
-        if (!EmailValidator.getInstance().isValid(email.receiverAddress)) {
-            return createBadRequest("Invalid email address")
+    override fun checkoutFinlandCancel(): Response {
+        TODO("Not yet implemented")
+    }
+
+    @RolesAllowed(value = [UserRole.PATIENT.name])
```

Comment:
Callback url may not be authorized because it's coming from checkout finland, it has to be completely unauthorized.

Authorization is done by HMAC calculation